Payconiq removal

JustNireon opened the pull request:
Resolves #900,

This removes all references to Payconiq in the codebase (where possible)
This also removes the dropdowns that once showed the choice between Ideal and Payconiq, but are now useless.
This should be tested thoroughly on staging, as test payments cannot go through in development (due to the return address not being reachable by the payment provider).

Riscky commented:
I checked, and I'm pretty sure we don't have to change anything in the database schema.
IMHO we can put this on staging and start testing

Riscky commented:
I rebased on the latest master, fixed the CI failures (linting, brakeman), removed some more implicit references to  Payconiq and updated the OP.
I think the PR should be ready for testing now

Raymonzut commented on app/models/payment.rb:
```diff
@@ -4,23 +4,23 @@ class Payment < ApplicationRecord
 
   enum status: { failed: 0, in_progress: 1, successful: 2 }
-  enum payment_type: { ideal: 0, payconiq_online: 1, payconiq_display: 2, pin: 3 }
+  enum payment_type: { ideal: 0, pin: 3 }
```
Good to see the variants still retaining their original numbers for backwards-compatibility,  but be careful

Raymonzut commented on db/seeds/payments.rb:
```diff
@@ -7,7 +7,7 @@
   15.times do
     transactiontype = Faker::Number.within(range: 0..1)
-    paymenttype = Faker::Number.within(range:0..3)
+    paymenttype = Faker::Number.within(range:0..1)
```
, as paymenttype variants are 0 and 3, not 0 and 1 ;D

Raymonzut commented on db/seeds/payments.rb:
```diff
@@ -7,7 +7,8 @@
     transactiontype = Faker::Number.within(range: 0..1)
-    paymenttype = Faker::Number.within(range:0..3)
+    # either ideal or pin (see app/models/payment.rb)
+    paymenttype = Faker::Number.within(range: 0.0..1.0) < 0.5 ? 0 : 3
```
which you did correctly here :+1:

Raymonzut commented on app/models/payment.rb:
```diff
@@ -190,10 +139,7 @@ def transaction_fee
-      return Settings.payconiq_transaction_costs
-    else
-      # :pin, :payconiq_display
+    when :pin
```
Shouldn't it also be `0` on `:payconiq_online` till it is removed, as it could still be present in the database?

Riscky replied:
good point, I'll add that

Raymonzut reviewed (changes requested):
Generally looks good, I propose the following to discuss:
- [ ] Seeds: Reconsider the Faker generators for `paymenttype`
- [x] Transaction fees: will `:payconiq_online` still need to be accounted for in the fees, as they could be in the old DB?

Riscky commented:
I have updated the fees calculation. The seeds generator is already correct I think, your comment is on an outdated version of the code.

Raymonzut reviewed (dismissed):
LGTM, all pointers from last review were addressed and added :D

- seeds => has been fixed since, now outdated
- fees calculation: corrected

JustNireon commented:
Thanks for the help! I've been very busy so the fixes are appreciated

On Wed, 6 Apr 2022, 18:25 Raymon Zutekouw, ***@***.***> wrote:

> ***@***.**** approved this pull request.
>
> LGTM, all pointers from last review were addressed and added :D
>
>    - seeds => has been fixed since, now outdated
>    - fees calculation: corrected
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/svsticky/constipated-koala/pull/907#pullrequestreview-933770905>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AC7C7VDQ72V5JFEDBVT6NF3VDW3GNANCNFSM5HYZW3FA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

Riscky commented:
Thanks for the review!
